examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -112,83 +117,86 @@ def load_mtpl2(n_samples=100000):
 
 ##############################################################################
-# It is worth noting that 92 % of policyholders have zero claims, and if we
-# were to convert this problem into a binary classification task, it would be
-# significantly imbalanced.
+# A constant prediction baseline
+# ------------------------------
+#
+# It is worth noting that more than 93% of policyholders have zero claims. If
+# we were to convert this problem into a binary classification task, it would
+# be significantly imbalanced.
 #
 # To evaluate the pertinence of the used metrics, we will consider as a
 # baseline a "dummy" estimator that constantly predicts the mean frequency of
 # the training sample.
 
-df_train, df_test = train_test_split(df, random_state=0)
+from sklearn.dummy import DummyRegressor
+from sklearn.pipeline import Pipeline
+from sklearn.model_selection import train_test_split
 
-dummy = make_pipeline(
-    linear_model_preprocessor,
-    DummyRegressor(strategy='mean')
-)
-dummy.fit(df_train, df_train["Frequency"],
-          dummyregressor__sample_weight=df_train["Exposure"])
+df_train, df_test = train_test_split(df, test_size=0.33, random_state=0)
+
+dummy = Pipeline([
+    ("preprocessor", linear_model_preprocessor),
+    ("regressor", DummyRegressor(strategy='mean')),
+]).fit(df_train, df_train["Frequency"],
+       regressor__sample_weight=df_train["Exposure"])
+
+
+##############################################################################
+# Let's compute the performance of this constant prediction baseline with 3
+# different regression metrics:
+
+from sklearn.metrics import mean_squared_error
+from sklearn.metrics import mean_absolute_error
+from sklearn.metrics import mean_poisson_deviance
 
 
 def score_estimator(estimator, df_test):
     """Score an estimator on the test set."""
-
     y_pred = estimator.predict(df_test)
 
     print("MSE: %.3f" %
           mean_squared_error(df_test["Frequency"], y_pred,
-                             df_test["Exposure"]))
+                             sample_weight=df_test["Exposure"]))
     print("MAE: %.3f" %
           mean_absolute_error(df_test["Frequency"], y_pred,
-                              df_test["Exposure"]))
+                              sample_weight=df_test["Exposure"]))
 
     # ignore non-positive predictions, as they are invalid for
     # the Poisson deviance
     mask = y_pred > 0
     if (~mask).any():
-        warnings.warn("Estimator yields non-positive predictions for {} "
-                      "samples out of {}. These will be ignored while "
-                      "computing the Poisson deviance"
-                      .format((~mask).sum(), mask.shape[0]))
+        n_clipped, n_samples = (~mask).sum(), mask.shape[0]
+        print(f"WARNING: Estimator yields non-positive predictions for "
+              f"{n_clipped} samples out of {n_samples}. These predictions "
+              f"will be clipped while computing the Poisson deviance.")
```

Comment:
There is a text section below that still says that the samples are ignored instead of clipped (currently line 204)

Reply:
As discusses in https://github.com/scikit-learn/scikit-learn/pull/14300#discussion_r319768664, it is not a good idea to clip because the result depends on the size of the "epsilon". I'd propose to revert to ignoring samples with predictions <= 0.

Reply:
Alright. This does not change the relative order of the models: the Poisson deviance of the Poisson GLM and Poisson GBRT models are still lower than the Poisson deviance computed on the masked predictions of the Ridge model.

```
Constant mean frequency evaluation:
MSE: 0.564
MAE: 0.189
mean Poisson deviance: 0.625
Ridge evaluation:
MSE: 0.560
MAE: 0.177
WARNING: Estimator yields non-positive predictions for 1315 samples out of 223745. These predictions will be ignored while computing the Poisson deviance.
mean Poisson deviance: 0.601
PoissonRegressor evaluation:
MSE: 0.560
MAE: 0.186
mean Poisson deviance: 0.594
Poisson Gradient Boosted Trees evaluation:
MSE: 0.556
MAE: 0.183
mean Poisson deviance: 0.573
```

Reply:
@ogrisel Thank you. My feeling is that the clipping is just plain wrong, at least dirty, and would have motivated others to do so.

Reply:
No problem.